compiler-rt/lib/asan/asan_interceptors.cpp

```diff
@@ -47,6 +47,12 @@
 #    define ASAN_PTHREAD_CREATE_VERSION "GLIBC_2.2"
 #  endif
 
+#if !defined(MAP_FAILED)
+#  define MAP_FAILED ((void *)-1)
```

Comment:
`map_fixed` (in lowercase) is defined in compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp for this purpose (example usage: compiler-rt/lib/hwasan/hwasan_interceptors.cpp).